functions/src/emailUnsubscribe/index.js

```diff
@@ -0,0 +1,62 @@
+import * as functions from 'firebase-functions';
+import * as admin from 'firebase-admin';
+import { USERS_COLLECTION } from 'constants/firestorePaths';
+import { to } from 'utils/async';
+
+/**
+ * Handle request from calling /emailUnsubscribe endpoint
+ * @param {object} req - Express HTTP Request
+ * @param {object} res - Express HTTP Response
+ * @returns {Promise} Resolves after handling request
+ */
+async function emailUnsubscribeRequest(req, res) {
+  // Write response to request to end function execution
+  const { body, query } = req;
+  console.log('Unsubscribe request received', { body, query });
+  if (!query) {
+    console.error('Query parameters not included no request, exiting');
+    return res.status(500).send('Error');
+  }
+  const { uid } = req.query;
+  console.log(`Loading account for user: ${uid}`);
+  // Get user account to confirm it exists
+  const [userGetErr, userSnap] = await to(
+    admin.firestore().doc(`${USERS_COLLECTION}/${uid}`).get(),
+  );
+
+  // Exit if there is an error getting the user account
+  if (userGetErr) {
+    console.error(`Error getting user account for id: ${uid}`, userGetErr);
+    return res.status(500).send('Error');
+  }
+
+  // Exit if user account not found
+  if (!userSnap.exists) {
+    console.log(`User not found for id: ${uid}`);
+    return res.status(500).send('Error');
+  }
+
+  // Disable email notifications on the user's account
+  const [updateErr] = await to(
+    userSnap.ref.update({ emailNotifications: false }),
```

Comment:
No, you shouldn't be able to update someone else's user account? It is on their private user object which can only be edited by the matching UID?

Reply:
Maybe I am misunderstanding something.  Can't the param be overridden since we are using `req.query` instead of `req.auth.uid`?

Reply:
The `req.query` will just be the email that was used to send to the user, so you would actually need to know the email of the user in order to unsubscribe them. In the near future we could add a token that is generated per email address so that only someone with both token and email of a user could disable notifications.

We won't have `req.auth.uid` since this link will be getting called from an email, thats the reason I went with req.query to begin with. All great questions!